internal/provider/integration_gitlab_resource.go

```diff
@@ -0,0 +1,302 @@
+
+	// Save data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *integrationGitlabResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var data integrationGitlabResourceModel
+
+	// Read Terraform prior state data into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Read API call logic
+
+	// Save updated data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *integrationGitlabResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var data integrationGitlabResourceModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	gitlabType := mondoov1.GitlabIntegrationType("NONE")
+	if data.Group.ValueString() != "" {
+		gitlabType = mondoov1.GitlabIntegrationType("GROUP")
+	}
+
+	opts := mondoov1.ClientIntegrationConfigurationInput{
+		GitLabConfigurationOptions: &mondoov1.GitlabConfigurationOptionsInput{
+			Type:                 gitlabType,
+			Group:                mondoov1.NewStringPtr(mondoov1.String(data.Group.ValueString())),
+			BaseURL:              mondoov1.NewStringPtr(mondoov1.String(data.BaseURL.ValueString())),
+			DiscoverGroups:       mondoov1.NewBooleanPtr(mondoov1.Boolean(data.Discovery.Groups.ValueBool())),
+			DiscoverProjects:     mondoov1.NewBooleanPtr(mondoov1.Boolean(data.Discovery.Projects.ValueBool())),
+			DiscoverTerraform:    mondoov1.NewBooleanPtr(mondoov1.Boolean(data.Discovery.Terraform.ValueBool())),
+			DiscoverK8sManifests: mondoov1.NewBooleanPtr(mondoov1.Boolean(data.Discovery.K8sManifests.ValueBool())),
+			Token:                mondoov1.NewStringPtr(mondoov1.String(data.Credential.Token.ValueString())),
+		},
+	}
+	// Update API call logic
+	_, err := r.client.UpdateIntegration(ctx,
+		data.Mrn.ValueString(),
+		data.Name.ValueString(),
+		mondoov1.ClientIntegrationTypeGitLab,
+		opts,
+	)
+	if err != nil {
+		resp.Diagnostics.
+			AddError("Client Error",
+				fmt.Sprintf("Unable to update GitLab integration, got error: %s", err),
+			)
+		return
+	}
+
+	// Save updated data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *integrationGitlabResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var data integrationGitlabResourceModel
+
+	// Read Terraform prior state data into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Delete API call logic
+	_, err := r.client.DeleteIntegration(ctx, data.Mrn.ValueString())
+	if err != nil {
+		resp.Diagnostics.
+			AddError("Client Error",
+				fmt.Sprintf("Unable to delete GitLab integration, got error: %s", err),
+			)
+		return
+	}
+}
```

Comment:
I don't see any `ImportState()`, do you have cycles to implement it?

Reply:
hab das gerade noch abgeklärt und werde die ImportState() Funktion noch bei allen PRs implementieren

Reply:
ImportState() is now implemented